storage/indexer_test.go

```diff
@@ -86,22 +86,128 @@ func BenchmarkIndexerGet(b *testing.B) {
 	})
 }
 
-func TestGet_ListAllPackages(t *testing.T) {
+func TestGet_ListPackages(t *testing.T) {
 	// given
 	fs := PrepareFakeServer(t, "testdata/search-index-all-full.json")
 	defer fs.Stop()
 	storageClient := fs.Client()
 	indexer := NewIndexer(util.NewTestLogger(), storageClient, FakeIndexerOptions)
 
-	err := indexer.Init(context.Background())
+	ctx := context.Background()
+	err := indexer.Init(ctx)
 	require.NoError(t, err, "storage indexer must be initialized properly")
 
-	// when
-	foundPackages, err := indexer.Get(context.Background(), &packages.GetOptions{})
+	cases := []struct {
+		name     string
+		options  *packages.GetOptions
+		expected int
+	}{
+		{
+			name:     "all packages filter nil",
+			options:  &packages.GetOptions{},
+			expected: 1133,
+		},
+		{
+			name: "all versions of packages including prerelease",
+			options: &packages.GetOptions{
+				Filter: &packages.Filter{
+					AllVersions: true,
+					Prerelease:  true,
+				},
+			},
+			expected: 1133,
+		},
+		{
+			name: "latest versions of packages not including prerelease",
+			options: &packages.GetOptions{
+				Filter: &packages.Filter{
+					AllVersions: false,
+					Prerelease:  false,
+				},
+			},
+			expected: 99,
+		},
+		{
+			name: "all packages with all versions and no prerelease",
+			options: &packages.GetOptions{
+				Filter: &packages.Filter{
+					AllVersions: true,
+				},
+			},
+			expected: 494,
+		},
+		{
+			name: "all packages with latest versions and no prerelease",
+			options: &packages.GetOptions{
+				Filter: &packages.Filter{
+					Prerelease: false,
+				},
+			},
+			expected: 99,
+		},
+		{
+			name: "all packages prerelease",
+			options: &packages.GetOptions{
+				Filter: &packages.Filter{
+					Prerelease: true,
+				},
+			},
+			expected: 147,
+		},
+		{
+			name: "all zeek packages with prerelease",
+			options: &packages.GetOptions{
+				Filter: &packages.Filter{
+					AllVersions: true,
+					Prerelease:  true,
+					PackageName: "zeek",
+				},
+			},
+			expected: 17,
+		},
+		{
+			name: "all packages with all versions of a giventype",
+			options: &packages.GetOptions{
+				Filter: &packages.Filter{
+					AllVersions: true,
+					Prerelease:  true,
+					PackageType: "solution",
+				},
+			},
+			expected: 2,
+		},
+		{
+			name: "one package of a giventype",
+			options: &packages.GetOptions{
+				Filter: &packages.Filter{
+					Prerelease:     true,
+					PackageName:    "tomcat",
+					PackageVersion: "0.3.0",
+				},
+			},
+			expected: 1,
+		},
+		{
+			name: "unknown package",
+			options: &packages.GetOptions{
+				Filter: &packages.Filter{
+					PackageName: "qwertyuiop",
+					PackageType: "integration",
+				},
+			},
+			expected: 0,
+		},
```

Comment:
This test case replaces the test `TestGet_UnknownPackage`.

Reply:
Maybe we can replace most tests in this file if we also add assertions for one package name and version :slightly_smiling_face: 

The tests that expect a package could be replaced by some case like:
```
		{
			name: "latest package",
			options: &packages.GetOptions{
				Filter: &packages.Filter{
					PackageName: "apm",
					PackageType: "integration",
				},
			},
			expected: 1,
			expectedName: "apm",
			expectedVersion: "8.2.0",
		},
```

Reply:
Right! I'll update this test to include that case (and remove the other).